syntax = "proto3";


service LLMService {
  // Chamada un√°ria para health check
  rpc HealthCheck (HealthRequest) returns (HealthResponse) {}
  
  // Stream de resposta da LLM
  rpc GenerateText (PromptRequest) returns (stream TextResponse) {}
  
  // Chamada para carregar modelos
  rpc LoadModel (ModelRequest) returns (ModelResponse) {}
}

// Mensagens
message HealthRequest {}

message HealthResponse {
  bool ready = 1;
  string model = 2;
}

message PromptRequest {
  string prompt = 1;
  optional int32 max_tokens = 2;
  optional float temperature = 3;
  optional string model = 4;
}

message TextResponse {
  string text = 1;
  bool is_final = 2;
  optional string error = 3;
}

message ModelRequest {
  string model_name = 1;
}

message ModelResponse {
  bool success = 1;
  string message = 2;
}